Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
+						typename IfcEntityTypesT::IfcBSplineSurfaceForm surfaceForm = surface->SurfaceForm;
 
-						if(cylindrical_surface) {
-							// Get radius.
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcPositiveLengthMeasure>& cylindrical_radius =
-								cylindrical_surface->Radius;*/
-							double cylindrical_radius = cylindrical_surface->Radius;
-							// TODO: implement		// Interpret values and calculate.
-						}
+						// Get attributes 5-7. For information only.
+						typename IfcEntityTypesT::IfcLogical uClosed = surface->UClosed;
+						typename IfcEntityTypesT::IfcLogical vClosed = surface->VClosed;
+						typename IfcEntityTypesT::IfcLogical selfIntersect = surface->SelfIntersect;
+					}
+					*/
 
-						// (2/4) IfcPlane SUBTYPE of IfcElementarySurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcPlane> elementary_surface_plane =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcPlane>(elementary_surface);
+					throw oip::UnhandledException(surface);
 
-						if(elementary_surface_plane) {
-							//  1----0     create big rectangular plane
-							//  |    |     ^ y
-							//  |    |     |
-							//  2----3     ---> x
-							{
-								double plane_span = HALF_SPACE_BOX_SIZE;
-								polyline_data->beginPolyline();
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(plane_span, plane_span, 0.0));
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(-plane_span, plane_span, 0.0));
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(-plane_span, -plane_span, 0.0));
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(plane_span, -plane_span, 0.0));
-
-								polyline_data->addPolylineIndex(0);
-								polyline_data->addPolylineIndex(1);
-								polyline_data->addPolylineIndex(2);
-								polyline_data->addPolylineIndex(3);
-							}
-							return;
-						}
+				}
 
-						// (3/4) IfcSphericalPlane SUBTYPE of IfcElementarySurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcSphericalSurface> spherical_surface =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcSphericalSurface>(elementary_surface);
+				/*! \brief  Converts \c IfcCurveBoundedPlane to ...
+					\param	surface	\c IfcCurveBoundedPlane entity to be interpreted.
+					\param	pos
+					\param	polylineData
+					\note	The \c IfcCurveBoundedPlane is a subtype of \c IfcBoundedSurface.
+				*/
 
-						if(spherical_surface) {
-							// Get radius.
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcPositiveLengthMeasure>& spherical_radius =
-								spherical_surface->m_Radius;*/
+				void convertIfcCurveBoundedPlane(const EXPRESSReference<typename IfcEntityTypesT::IfcCurveBoundedPlane>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {
 

```

Comment:
Same, also output param pos should be returned. See https://github.com/tumcms/Open-Infra-Platform/blob/75a2c0b4fda902bd97785cacb2f7498ad235fd6c/Core/src/IfcGeometryConverter/PlacementConverter.h#L80-L83

Reply:
Similar to above, `pos` is not a return parameter, rather it is an input. `std::shared_ptr<carve::input::PolylineSetData>` is the return value.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1